runtime/dynstats.c

```diff
@@ -609,8 +918,22 @@ dynstats_inc(dynstats_bucket_t *b, uchar* metric) {
 	}
 
 	if (ctr == NULL) {
-		CHKiRet(dynstats_addNewCtr(b, metric, 1));
+		CHKiRet(dynstats_addNewCtr(b, metric, 1, 0));
+	}
+
+	time_t now;
+	datetime.GetTime(&now);
+	if (dynstats_shouldPersist(b, now)) {
+		/* TODO: suggested to be refactored into bg thread */
+		persistBucketState(b);
+		if (b->persist_state_interval) {
```

Comment:
We need a better name for this. `persist_state_interval` doesn't really help one understand what this field is used for.

Reply:
yes, looking back, the idea was to snap to the imfile config naming convention: https://www.rsyslog.com/doc/v8-stable/configuration/modules/imfile.html#persiststateinterval
are you ok with leaving the configuration name as `persistStateInterval` ? We could perhaps rename the variable to something like `persistStateWriteCountInterval` or something similar?